app/connectors/aws/S3EventParser.scala

```diff
@@ -67,21 +67,9 @@ class S3EventParser @Inject()(implicit ec: ExecutionContext) extends MessagePars
   private def interpretS3EventMessage(result: S3EventNotification): Future[FileUploadEvent] =
     result.records match {
       case Seq(S3EventNotificationRecord(_, "aws:s3", _, _, ObjectCreatedEventPattern(), s3Details)) =>
-        import org.slf4j.MDC
-
         val event = FileUploadEvent(S3ObjectLocation(s3Details.bucketName, s3Details.objectKey))
-        // Can't rely on MdcLoggingExecutionContext to add file-reference to the MDC, in the code below,
-        // because we don't already have the file-reference value when the thread begins executing.
-        // Therefore add file-reference to the MDC manually. This should be the only part of the code where we need to
-        // manually touch the MDC, since subsequent log statements will have access to the file-reference and can thus
-        // use MdcLoggingExecutionContext / LoggingDetails.
-        try {
-          MDC.put("file-reference", event.location.objectKey)
-          logger.debug(s"Created FileUploadEvent for Key=[${event.location.objectKey}].")
-          Future.successful(event)
-        } finally {
-          MDC.remove("file-reference")
-        }
+        logger.debug(s"Created FileUploadEvent for bucket=[${event.location.bucket}] object=[${event.location.objectKey}].")
```

Comment:
Can you access `withLoggingDetails` here?

Reply:
Following a recent change, we could now use `withLoggingDetails`  here, but the MDC could only contain the `objectKey` (which is already logged in the message itself).  It cannot contain the `file-reference` as this is not contained in the event message - it has to be retrieved from metadata - which can only happen after this event parsing has completed.  The `objectKey` is not typically added to the MDC elsewhere - support was only added to retain some context for any exception thrown prior to the metadata being retrieved.